azure-core/tests/test_universal_pipeline.py

```diff
@@ -0,0 +1,164 @@
+# of this software and associated documentation files (the ""Software""), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+#--------------------------------------------------------------------------
+try:
+    from unittest import mock
+except ImportError:
+    import mock
+
+import requests
+
+import pytest
+
+from azure.core.exceptions import DecodeError
+from azure.core.configuration import Configuration
+from azure.core.pipeline import (
+    PipelineResponse,
+    PipelineRequest,
+)
+from azure.core.pipeline.transport import (
+    HttpRequest,
+    HttpResponse,
+)
+from azure.core.pipeline.transport import RequestsTransportResponse
+
+from azure.core.pipeline.policies.universal import (
+    NetworkTraceLoggingPolicy,
+    ContentDecodePolicy,
+    UserAgentPolicy
+)
+
+def test_user_agent():
+
+    with mock.patch.dict('os.environ', {'AZURE_HTTP_USER_AGENT': "mytools"}):
+        policy = UserAgentPolicy(None)
+        assert policy.user_agent.endswith("mytools")
+
+        request = HttpRequest('GET', 'http://127.0.0.1/')
+        policy.on_request(PipelineRequest(request))
+        assert request.headers["user-agent"].endswith("mytools")
+
+@mock.patch('azure.core.pipeline.policies.universal._LOGGER')
+def test_no_log(mock_http_logger):
+    universal_request = HttpRequest('GET', 'http://127.0.0.1/')
+    request = PipelineRequest(universal_request)
+    http_logger = NetworkTraceLoggingPolicy()
+    response = PipelineResponse(request, HttpResponse(universal_request, None))
+
+    # By default, no log handler for HTTP
+    http_logger.on_request(request)
+    mock_http_logger.debug.assert_not_called()
+    http_logger.on_response(request, response)
+    mock_http_logger.debug.assert_not_called()
+    mock_http_logger.reset_mock()
+
+    # I can enable it per request
+    http_logger.on_request(request, **{"enable_http_logger": True})
+    assert mock_http_logger.debug.call_count >= 1
+    http_logger.on_response(request, response, **{"enable_http_logger": True})
+    assert mock_http_logger.debug.call_count >= 1
+    mock_http_logger.reset_mock()
```

Comment:
Are both calls to `on_response` individually guaranteed to raise the call count by at least 1? Is so, that information seems lost by comparing only to `>=1` in succession. Could `reset_mock()` be called in between assertions?

Reply:
True - fixed!